ansible_inventory_server/utils.py

```diff
@@ -46,3 +46,30 @@ def json_response(self, data):
             response = json.dumps(data)
 
         self.write(response)
+
+
+def _ip_bits(ip):
+    """'127.0.0.1'  -->  '01111111000000000000000000000001'"""
+    ip = ip.split('/')[0]
+    return ''.join(bin(int(byte))[2:].rjust(8, '0') for byte in ip.split('.'))
+
+
+def ip_in_subnet(ip, subnet):
+    subnet_ip, length = subnet.split('/')
+    return _ip_bits(ip)[:int(length)] == _ip_bits(subnet_ip)[:int(length)]
+
+
+def filter_ip_addresses(ip_addresses, kwargs):
+    """Given a list of @ip_addresses, choose the addresses  @kwargs['subnet'].
+    If there are not any, then return the first ip_address"""
+
+    subnet = kwargs.get('subnet')
+    try:
+        result = [ip for ip in ip_addresses if ip_in_subnet(ip, subnet)]
+    except (TypeError, ValueError, IndexError, AttributeError):
```

Comment:
by using the check mentioned above this can be refactored to check if ips and subnet are valid (no ValueError) and return the "overlapping" ips